packages/nimble-components/src/combobox/index.ts

```diff
@@ -340,13 +341,11 @@ export class Combobox
 
         const filter = this.filter.toLowerCase();
 
-        this.filteredOptions = this._options.filter(o => o.text.toLowerCase().startsWith(filter));
+        this.filteredOptions = this._options.filter(
+            o => o.text.toLowerCase().startsWith(filter) && !o.hidden
+        );
 
         if (this.isAutocompleteList) {
-            if (!this.filteredOptions.length && !filter) {
-                this.filteredOptions = this._options;
-            }
-
```

Comment:
`filter` should not be `null`/`undefined`, so `!filter` should mean `filter === ''`. When the filter is empty, `startsWith('')` will let every option through, so the only way to get in this block is if there are no visible (i.e. not `hidden`) options. We don't need to reset it to `this._options`, and if all the options are `hidden`, resetting it is actually wrong.